date_range/models/date_range.py

```diff
@@ -28,11 +28,45 @@ def _default_company(self):
     active = fields.Boolean(
         help="The active field allows you to hide the date range without "
         "removing it.", default=True)
+    parent_type_id = fields.Many2one(
+        related='type_id.parent_type_id',
+        store=True,
+        readonly=True)
+    parent_id = fields.Many2one(
+        comodel_name='date.range', string="Parent",
+        domain="['|', ('type_id.parent_type_id', '!=', parent_type_id), "
```

Comment:
No need to have the domain as a string. String domains are evaluated client side, if possible it is better to evaluate server side. Also I think the type_id of the parent should be equal (if filled) as the parent_type_id, not unequal.

Reply:
Shall this domain be re-introduced in the view, then?

Reply:
Maybe we have a misunderstanding. I did not suggest to remove the domain, I said you can add it as a proper python array, so without the enclosing double quotes.